From 92acb462ef15647780c145888628538e7c238bfa Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 22 Feb 2021 20:35:41 +0200
Subject: [PATCH] [SW-20154] skip clearing the top_rom_valid entries

Goya2 cleared both the TOP_ROM_VALID registers during
JTAG disable and set both to 0xFFFFFFFF during JTAG
enable. This setting is not required for Gaudi2. Hence
move this to platform specific code.

Change-Id: I30fb659e910ccd7dcdcb7e92cf74ec4711e66efe
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 7da328c..eeb46c8 100644
--- a/common/security.c
+++ b/common/security.c
@@ -11,22 +11,6 @@
 #include "security.h"
 #include "spi_flash.h"
 
-void security_disable_jtag_if(void)
-{
-	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
-	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
-	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
-	security_disable_jtag_mstr_if();
-}
-
-void security_enable_jtag_if(void)
-{
-	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
-	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
-	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
-	security_enable_jtag_mstr_if();
-}
-
 void security_debug_if_secure(void)
 {
 	/* Configure SMI,I2C and JTAG to be "secure" */
diff --git a/gaudi2_arc_gnu/security.c b/gaudi2_arc_gnu/security.c
index aa32271..77e1da9 100644
--- a/gaudi2_arc_gnu/security.c
+++ b/gaudi2_arc_gnu/security.c
@@ -567,3 +567,15 @@
 			EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0)
 		security_enable_twi_if();
 }
+
+void security_disable_jtag_if(void)
+{
+	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
+	security_disable_jtag_mstr_if();
+}
+
+void security_enable_jtag_if(void)
+{
+	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
+	security_enable_jtag_mstr_if();
+}
diff --git a/goya2_arc/security.c b/goya2_arc/security.c
index 85acee3..1ac70fe 100644
--- a/goya2_arc/security.c
+++ b/goya2_arc/security.c
@@ -479,3 +479,18 @@
 		security_disable_twi_if();
 }
 
+void security_disable_jtag_if(void)
+{
+	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
+	regmap_writel(0, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
+	security_jtag_dap(SECURITY_JTAG_DAP_DISABLE);
+	security_disable_jtag_mstr_if();
+}
+
+void security_enable_jtag_if(void)
+{
+	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_0);
+	regmap_writel(0xFFFFFFFF, mmPSOC_CS_TRACE_TOP_ROM_VALID_1);
+	security_jtag_dap(SECURITY_JTAG_DAP_ENABLE);
+	security_enable_jtag_mstr_if();
+}
